app/src/main/java/org/schabi/newpipe/player/ui/MainPlayerUi.java

```diff
@@ -0,0 +1,988 @@
+            return true;
+        }
+        return super.onKeyDown(keyCode);
+    }
+    //endregion
+
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Video size, orientation, fullscreen
+    //////////////////////////////////////////////////////////////////////////*/
+    //region Video size, orientation, fullscreen
+
+    private void setupScreenRotationButton() {
+        binding.screenRotationButton.setVisibility(globalScreenOrientationLocked(context)
+                || isVerticalVideo || DeviceUtils.isTablet(context)
+                ? View.VISIBLE : View.GONE);
+        binding.screenRotationButton.setImageDrawable(AppCompatResources.getDrawable(context,
+                isFullscreen ? R.drawable.ic_fullscreen_exit
+                        : R.drawable.ic_fullscreen));
+    }
+
+    @Override
+    public void onVideoSizeChanged(@NonNull final VideoSize videoSize) {
+        super.onVideoSizeChanged(videoSize);
+        isVerticalVideo = videoSize.width < videoSize.height;
+
+        if (globalScreenOrientationLocked(context)
+                && isFullscreen
+                && isLandscape() == isVerticalVideo
+                && !DeviceUtils.isTv(context)
+                && !DeviceUtils.isTablet(context)) {
+            // set correct orientation
+            player.getFragmentListener().ifPresent(
+                    PlayerServiceEventListener::onScreenRotationButtonClicked);
+        }
+
+        setupScreenRotationButton();
+    }
+
+    public void toggleFullscreen() {
+        if (DEBUG) {
+            Log.d(TAG, "toggleFullscreen() called");
+        }
+        final PlayerServiceEventListener fragmentListener
+                = player.getFragmentListener().orElse(null);
+        if (fragmentListener == null || player.exoPlayerIsNull()) {
+            return;
+        }
+
+        isFullscreen = !isFullscreen;
+        if (isFullscreen) {
+            // Android needs tens milliseconds to send new insets but a user is able to see
+            // how controls changes it's position from `0` to `nav bar height` padding.
+            // So just hide the controls to hide this visual inconsistency
+            hideControls(0, 0);
+        } else {
+            // Apply window insets because Android will not do it when orientation changes
+            // from landscape to portrait (open vertical video to reproduce)
+            binding.playbackControlRoot.setPadding(0, 0, 0, 0);
+        }
+        fragmentListener.onFullscreenStateChanged(isFullscreen);
+
+        if (isFullscreen) {
+            binding.titleTextView.setVisibility(View.VISIBLE);
+            binding.channelTextView.setVisibility(View.VISIBLE);
+            binding.playerCloseButton.setVisibility(View.GONE);
+        } else {
+            binding.titleTextView.setVisibility(View.GONE);
+            binding.channelTextView.setVisibility(View.GONE);
+            binding.playerCloseButton.setVisibility(View.VISIBLE);
+        }
```

Comment:
Same as above, I would also replace this by  ``binding.titleTextView.setVisibility(isFullscreen ? View.VISIBLE : View.GONE);``